### DISCUSSION TITLE
Deleted all shortcuts in IFS Browser and unable to reconnect to AS/400

### DISCUSSION THREAD

**Discussion by NicholasRicher** [codefori/vscode-ibmi#1425] [Q&A]
Hello,

I'm new to Code for IBM i and I accidentally deleted the default "/" root directory shortcut in the IFS Browser instead of simply removing it.  Now I have no shortcuts in my IFS Browser. I've sense disconnected from my AS/400 and upon trying to reconnect I get a "No response from server" error message. I've tried deleting the settings.json file for Code for IBM i, uninstalling/reinstalling the extension, and uninstalling/reinstalling VSCode but am still getting the error. I also tried resetting VSCode's global settings by deleting the .vscode folder and removing code-for-ibmi.connections from my User Settings json file.

Any help would be greatly appreciated.
Thanks.

**Comment by worksofliam** [comment]
Are you able to connect with other SSH clients or from the terminal using SSH?

**Reply by NicholasRicher** [reply]
Yes, I've still been able to login to my AS/400 with a 5250 emulator.

**Reply by NicholasRicher** [reply]
The Code for IBM i runtime status also displays an uncaught error, "Cannot read properties of undefined (reading 'name')" when I try to connect.

**Reply by chrjorgensen** [reply]
Let's keep it in one thread....

5250 is not SSH and Code for IBM i does not use 5250!

Please verify SSH is running on the server by connecting using another SSH client, different from Code for IBM i.

**Reply by NicholasRicher** [reply]
I don't feel comfortable using another SSH client. I've been manually starting SSH in green screen using STRTCPSVR SERVER(*SSHD). This is how I was able to connect to the server with Code for IBM i before I deleted the "/" shortcut.

**Reply by NicholasRicher** [reply]
I typed SSH myusername@my_ip_address in the command line and it says the connection is closed by port 22 after I started an SSH server using the above command.

**Reply by chrjorgensen** [reply]
@NicholasRicher If your SSH command fails as described, then it's a problem with the SSH connection, not Code for IBM i.
You could try and debug by running `ssh -vvv <myusername>@<my_ip_address>` (the `-vvv` option makes the ssh command very, very verbose). This may tell you why your SSH connection is rejected (server issue, user issue, authority issue).

When you get the SSH connection to work, then Code for IBM i will probably work also.

**Reply by NicholasRicher** [reply]
When I accidentally deleted "/" from the IFS Browser I believe it deleted the directories and files behind it. My QOpenSys directory is empty and the sshd_config file that was inside of it is missing.

**Reply by worksofliam** [reply]
@NicholasRicher so did you remove the shortcut or did you delete the directory? 

You usually can't delete directories without a confirmation window.

**Reply by NicholasRicher** [reply]
I deleted the directory and clicked yes on the confirmation window. You can probably tell I don't really know what I'm doing. Code for IBM i said that all of its features wouldn't be usable until I specified a home directory for my user profile. The IFS Browser was open by default and only had the root file directory listed underneath it. I assumed it was only a bookmarking tool for file paths on my AS/400 so without knowing the difference between remove shortcut and delete object I mistakenly chose delete object.

**Reply by worksofliam** [reply]
@NicholasRicher sorry to hear that it happened. I have been chatting to our internal team. We can't do much more other than warn the user about directories they're going to delete and have them confirm it. We are to try and add some additional checks to stop deletion of the root in the future.

**Reply by NicholasRicher** [reply]
Do you know of any other directories that could've been deleted? I haven't found any other ones besides QOpenSys.

**Reply by worksofliam** [reply]
@NicholasRicher if you tried to delete root, it would try and delete everything below that that your user would have authority too and would stop when it fails to delete at any point.

**Reply by NicholasRicher** [reply]
What would you do to try to restore it? Can you offer any ideas? I really appreciate your timely communication.

**Reply by worksofliam** [reply]
@NicholasRicher honestly I would consider looking into a backup. If you use the IFS in any production manner I'd look into checking those objects/directories/files exist and if they don't restore them from backup.

**Reply by worksofliam** [reply]
@NicholasRicher if you need emergency help (non-IBM vendor service) shoot me a DM on LinkedIn and I can put you in contact with them. Thanks!

https://www.linkedin.com/in/worksofbarry

**Reply by NicholasRicher** [reply]
Thank you very much. I believe this issue can be closed.

**Comment by chrjorgensen** [comment]
Code for IBM i uses SSH as transport method, so please check SSH is available on the server by connecting using an SSH client, e g. Putty.

**Reply by NicholasRicher** [reply]
Yes I've still been able to connect to my AS/400 with a 5250 emulator. The Code for IBM i runtime status also displays an uncaught error, "Cannot read properties of undefined (reading 'name')" when I try to connect.
